x/cork/keeper/keeper.go

```diff
@@ -222,6 +359,56 @@ func (k Keeper) GetApprovedCorks(ctx sdk.Context, threshold sdk.Dec) (approvedCo
 	return winningCorks
 }
 
+func (k Keeper) GetApprovedScheduledCorks(ctx sdk.Context, currentBlockHeight uint64, threshold sdk.Dec) (approvedCorks []types.Cork) {
+
+	corksForBlockHeight := make(map[uint64][]types.Cork)
+	corkPowersForBlockHeight := make(map[uint64][]uint64)
+
+	totalPower := k.stakingKeeper.GetLastTotalPower(ctx)
+
+	k.IterateScheduledCorks(ctx, func(val sdk.ValAddress, scheduledBlockHeight uint64, addr common.Address, cork types.Cork) (stop bool) {
+		// only operate on scheduled corksForBlockHeight that are valid
+		if currentBlockHeight < scheduledBlockHeight {
```

Comment:
Based on the update methodology, shouldn't we only execute on exact block height matches?

Reply:
My understanding was that it was a "on or after" rule. Not sure why it would happen, but someone could scheduled a cork for the past, and it would be stuck forever if we didn't have this check